File: src/main/java/by/iba/repair_report/service/AssociationService.java
package by.iba.repair_report.service;

import by.iba.repair_report.dto.request.AssociationRequest;
import by.iba.repair_report.dto.response.AssociationResponse;

import java.util.List;

public interface AssociationService {
    AssociationResponse createAssociation(AssociationRequest request);
    AssociationResponse updateAssociation(Long id, AssociationRequest request);
    void deleteAssociation(Long id);
    AssociationResponse getAssociationById(Long id);
    List<AssociationResponse> getAllAssociations();
    List<AssociationResponse> searchAssociations(String name);
}

File: src/main/frontend/src/services/associationService.js
import api from './api';

export const associationService = {
    getAllAssociations: async () => {
        const response = await api.get('/associations');
        return response.data;
    },

    getAssociationById: async (id) => {
        const response = await api.get(`/associations/${id}`);
        return response.data;
    },

    createAssociation: async (associationData) => {
        const response = await api.post('/associations', associationData);
        return response.data;
    },

    updateAssociation: async (id, associationData) => {
        const response = await api.put(`/associations/${id}`, associationData);
        return response.data;
    },

    deleteAssociation: async (id) => {
        await api.delete(`/associations/${id}`);
    },

    searchAssociations: async (name) => {
        const response = await api.get(`/associations/search?name=${name}`);
        return response.data;
    }
};